tools/linker/CoreTypeMapStep.cs

```diff
@@ -20,37 +21,162 @@
 namespace MonoTouch.Tuner {
 
 	// This class is shared between Xamarin.Mac and Xamarin.iOS
-	public class CoreTypeMapStep : TypeMapStep {
-		HashSet<TypeDefinition> cached_isnsobject = new HashSet<TypeDefinition> ();
-		Dictionary<TypeDefinition, bool?> isdirectbinding_value = new Dictionary<TypeDefinition, bool?> ();
+	public class CoreTypeMapStep :
+#if NET
+		ConfigurationAwareStep
+#else
+		TypeMapStep
+#endif
+	{
+
+#if NET
+		protected override string Name { get; } = "CoreTypeMap";
+		protected override int ErrorCode { get; } = 2381;
```

Comment:
This should be a multiple of 10, because the base class may increase the actual error number to identify each specific error case.

```suggestion
		protected override int ErrorCode { get; } = 2390;
```